c_glib/parquet-glib/arrow-file-writer.cpp

```diff
@@ -590,7 +590,7 @@ gparquet_arrow_file_writer_new_row_group(GParquetArrowFileWriter *writer,
 {
   auto parquet_arrow_file_writer = gparquet_arrow_file_writer_get_raw(writer);
   return garrow::check(error,
-                       parquet_arrow_file_writer->NewRowGroup(chunk_size),
+                       parquet_arrow_file_writer->NewRowGroup(),
```

Comment:
Could you also remove the `chunk_size` parameter from this function?

Reply:
Sure, do I need to deprecate it?

Reply:
You don't need to deprecate it. We can do it as a breaking change. Because this will not be used by many users. This is a method added in 18.0.0.

```diff
diff --git a/c_glib/parquet-glib/arrow-file-writer.cpp b/c_glib/parquet-glib/arrow-file-writer.cpp
index 2b8e2bdeac..82f86e823b 100644
--- a/c_glib/parquet-glib/arrow-file-writer.cpp
+++ b/c_glib/parquet-glib/arrow-file-writer.cpp
@@ -574,7 +574,6 @@ gparquet_arrow_file_writer_write_table(GParquetArrowFileWriter *writer,
 /**
  * gparquet_arrow_file_writer_new_row_group:
  * @writer: A #GParquetArrowFileWriter.
- * @chunk_size: The max number of rows in a row group.
  * @error: (nullable): Return location for a #GError or %NULL.
  *
  * Start a new row group.
@@ -585,7 +584,6 @@ gparquet_arrow_file_writer_write_table(GParquetArrowFileWriter *writer,
  */
 gboolean
 gparquet_arrow_file_writer_new_row_group(GParquetArrowFileWriter *writer,
-                                         gsize chunk_size,
                                          GError **error)
 {
   auto parquet_arrow_file_writer = gparquet_arrow_file_writer_get_raw(writer);
diff --git a/c_glib/parquet-glib/arrow-file-writer.h b/c_glib/parquet-glib/arrow-file-writer.h
index 2c82f7c1f8..b35cea1182 100644
--- a/c_glib/parquet-glib/arrow-file-writer.h
+++ b/c_glib/parquet-glib/arrow-file-writer.h
@@ -136,7 +136,6 @@ gparquet_arrow_file_writer_write_table(GParquetArrowFileWriter *writer,
 GPARQUET_AVAILABLE_IN_18_0
 gboolean
 gparquet_arrow_file_writer_new_row_group(GParquetArrowFileWriter *writer,
-                                         gsize chunk_size,
                                          GError **error);
 
 GPARQUET_AVAILABLE_IN_18_0
```

Reply:
Could you also update `c_glib/parquet-glib/arrow-file-writer.h`?

BTW, could you use your fork not apache/arrow next time?

Reply:
Ohh, sorry. That was accidental.

Reply:
Created a new PR from my fork https://github.com/apache/arrow/pull/45088
Closing this and removing the branch.